STYLE_GUIDE.md

```diff
@@ -0,0 +1,190 @@
+
+## Github
+
+Development should following the regimented structure for branches and pull requests outlined herein.
+For further inspiration checkout https://nvie.com/posts/a-successful-git-branching-model/.
+
+ * Production branches
+   * Production repos never push to `master` or `develop` branches directly. Push-master can merge to these branches.
+   * `develop` should always pass Continuous Integration (CI), Continuous Delivery (CD) tests
+   * A release is a successful PR from the `develop` to `master`. The contents
+     from the PRs merged into develop forms the as basis for the release
+ * Feature branches
+   * should follow the naming convention `yourname/ghi-description`, where `ghi` is the github issue number
+   * all branches should be in lowercase, underscores/dashes are okay in branch names
+   * should have an open PR with the name "WIP: your feature description"
+   * All feature PRs should be opened for merge to the `develop` branch, unless stacked on other PR leading to `develop`.
+    * Ideally, for faster merges, break your PR into several PRs that stack on top of each other and lead to `develop`.
+ * Reference branches
+   * Sometimes it is necessary to keep a branch with reference information even there there is no intention of merging
+   * These branches should be named `yourname/reference-ghi-description`, where `ghi` is the github issue number
+ * Pull Requests (PRs)
+   * Do not merge PRs that aren't final -- e.g. no code dead code or code that won't be used should be committed.
+   * Always delete branch immediately after merge
+   * When there are to many commits, [Squash](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git)
+   them to keep the history clean - don't do this where commits ought to be kept for future reference.
+   * All pull requests should include any relevant additions to the `CHANGELOG.md`, if the PR is opened against the
+     develop branch then include the changes under an `unreleased changes` header at the top of the document.
+
+## Linters
+
+These must be applied to all (go) repos.
+
+ * [shellcheck](https://github.com/koalaman/shellcheck)
+ * [gometalinter (covers all important linters)](https://github.com/alecthomas/gometalinter)
+   - See the `Makefile` in each repo for ongoing linting progress. Running `make metalinter` is the ultimate goal, while `make metalinter_test` enables all the linters that are currently passing, for integration with CI.
+   * gometalinter may make use of the [Tendermint fork of golint](https://github.com/tendermint/lint/)
+
+## Various
+
+ * Reserve "Save" and "Load" for long-running persistence operations. When parsing bytes, use "Encode" or "Decode".
+ * Receiver variables should be at least 2 letters.  e.g. not `func (s \*Something) MyFunc(){}` but `func (sm *Something) MyFunc(){}`.  This makes refactoring easier when we rename Something to something else.
```

Comment:
Can we drop this? Code editors can rename symbols easily.

Reply:
Sure